lib/ratchet/html_parser.rb

```diff
@@ -0,0 +1,30 @@
+require 'ox'
+
+module Ratchet
+  class HtmlParser
+    def call(html)
+      document = Ox.parse(html)
+      parse(document)
+    end
+
+    private
+
+    def parse(element)
+      if element.is_a?(Ox::Element)
+        parse_element(element)
+      else
+        [:static, element]
+      end
+    end
+
+    def parse_element(element)
+      [
```

Comment:
The array syntax is a temple thing. They call these "abstractions". There are a number of existing ones that temple already knows how to deal with (https://github.com/judofyr/temple/blob/master/EXPRESSIONS.md).

Basically this is an abstract syntax tree. So from a high level, with temple, you:
1. Parse a temple into an initial abstraction. In my experience, this is often something specific to your domain. So in the case of Ratchet, we have "nuts" and "bolts" 😆 .
2. You pipe the parsed abstractions through a series of filters (in temple terminology). These are effectively functional transformations.
3. Finally, you run it through one final transformation called a generator. Temple comes with a few of these, but essentially this is the thing that concatenates strings.

Note: each of these: parser, filters, and generators are more generally referred to as "compilers" in temple terminology.

It's all quite nice! In fact it feels a little more complete than the Elixir equivalents I've worked with already. Too bad Ruby will never handle strings as fast as Elixir 😉